### PR TITLE
Registration backend

### PR DESCRIPTION
**Pull request by wwu3**
This is what use in Insomnia for testing backend.
image

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -14,6 +14,20 @@ const sequelize = new Sequelize('hamishgpt', 'user', 'root', {
   dialect: 'mysql'
 });
 
+app.post('/registration', function (request, response) {
+  return User.create({
```
Comment (Lestt) [inline server.js]: You have to check the presence of the mandatory fields and their validity (correct email address, not already taken,...)

Comment (Lestt) [reply server.js]: Also can you use the async await notation please ?

File: server.js [hunk 2]
```diff
@@ -14,6 +14,20 @@ const sequelize = new Sequelize('hamishgpt', 'user', 'root', {
   dialect: 'mysql'
 });
 
+app.post('/registration', function (request, response) {
+  return User.create({
+    username: request.body.username,
+    email: request.body.email,
+    password: request.body.password
+  }).then(function (users) {
```
Comment (Lestt) [inline server.js]: Also you are inserting one user or few users? Because this variable name is plural...

File: server.js (hunk 1 above)
Comment (wwu3) [reply server.js]: All fixed

File: server.js (hunk 2 above)
Comment (wwu3) [reply server.js]: All fixed

File: server.js [hunk 3]
```diff
@@ -14,6 +14,38 @@ const sequelize = new Sequelize('hamishgpt', 'user', 'root', {
...
+app.get('/ping', function(request, response) {
+  response.send("pong");
+});
+
+app.post('/registration', async (request, response) => {
+  const { username, email, password } = request.body;
+  if (username === null || username.length < 1) {
+    return response.status(400).send('Username is invalid');
+  }
+  if (email === null || email.length < 1) {
```
Comment (Lestt) [inline server.js:27]: You are not checking if it's a valid email address. You are just checking if the string for the email is at least 1 character.

**Review by Lestt — CHANGES_REQUESTED** [review]

File: server.js (hunk 3 above)
Comment (wwu3) [reply server.js:27]: fixed

**Pull request merged**